From aaa69a12966f8e9e3e81629b6b234fb95339601b Mon Sep 17 00:00:00 2001
From: Yannick Fertre <yannick.fertre@foss.st.com>
Date: Mon, 5 Jun 2023 22:51:08 +0200
Subject: [PATCH 0277/1141] arm64: defconfig: Enable DRM drivers for
 stm32mp257f boards

This sets DRM drivers. Default configuration uses them as builtin drivers
so that splash screen can be displayed earlier in the boot process.
In the package of enabled drivers there is :

	* DRM framework
	* DRM STM drivers
	* DRM DSI and STM DSI
	* STM LVDS
	* DSI and LVDS panels used from default config
	* Simple BACKLIGHT drivers
	* TOUCHSCREEN drivers for DSI and LVDS

Signed-off-by: Raphael Gallais-Pou <raphael.gallais-pou@foss.st.com>
Change-Id: Id7c51ec5e224d61be52aac2222dac99e969825d8
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/305991
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Tested-by: Yannick FERTRE <yannick.fertre@foss.st.com>
Reviewed-by: Yannick FERTRE <yannick.fertre@foss.st.com>
Domain-Review: Yannick FERTRE <yannick.fertre@foss.st.com>
---
 .../fragment-02-defconfig-addons.config       | 21 +++++++++++++++++++
 1 file changed, 21 insertions(+)

diff --git a/arch/arm64/configs/fragment-02-defconfig-addons.config b/arch/arm64/configs/fragment-02-defconfig-addons.config
index bc661d192e90..bab103f4b86c 100644
--- a/arch/arm64/configs/fragment-02-defconfig-addons.config
+++ b/arch/arm64/configs/fragment-02-defconfig-addons.config
@@ -2,16 +2,34 @@
 CONFIG_ARCH_STM32=y
 CONFIG_ARM_SCMI_CPUFREQ=y
 CONFIG_ARM_SMC_WATCHDOG=y
+CONFIG_BACKLIGHT_CLASS_DEVICE=y
+CONFIG_BACKLIGHT_GPIO=y
 CONFIG_CAN_M_CAN=m
 CONFIG_CAN_M_CAN_PLATFORM=m
 CONFIG_COUNTER=m
+CONFIG_DRM=y
+CONFIG_DRM_BRIDGE=y
+CONFIG_DRM_DW_MIPI_DSI=y
+CONFIG_DRM_KMS_HELPER=y
+CONFIG_DRM_MIPI_DSI=y
+CONFIG_DRM_PANEL=y
+CONFIG_DRM_PANEL_BRIDGE=y
+CONFIG_DRM_PANEL_LVDS=y
+CONFIG_DRM_PANEL_RAYDIUM_RM68200=y
+CONFIG_DRM_PANEL_SIMPLE=y
+CONFIG_DRM_SIMPLEDRM=y
+CONFIG_DRM_STM=y
+CONFIG_DRM_STM_DSI=y
+CONFIG_DRM_STM_LVDS=y
 CONFIG_DWMAC_GENERIC=y
 CONFIG_DWMAC_STM32=y
 CONFIG_DYNAMIC_DEBUG=y
+CONFIG_HDMI=y
 CONFIG_I2C_RPMSG=m
 CONFIG_I2C_STM32F7=y
 CONFIG_IIO_STM32_LPTIMER_TRIGGER=m
 CONFIG_IIO_STM32_TIMER_TRIGGER=m
+CONFIG_INPUT_TOUCHSCREEN=y
 CONFIG_JFFS2_FS=y
 CONFIG_MFD_STM32_LPTIMER=m
 CONFIG_MFD_STM32_TIMERS=m
@@ -65,6 +83,9 @@ CONFIG_STM32_RPROC=y
 CONFIG_STM32_TIMER_CNT=m
 CONFIG_STMMAC_ETH=y
 CONFIG_STMMAC_PLATFORM=y
+CONFIG_TOUCHSCREEN_GOODIX=y
+CONFIG_TOUCHSCREEN_ILI210X=y
 CONFIG_UBIFS_FS=y
 CONFIG_USB_DWC3_STM32=y
 CONFIG_USB_STM32_USBH=y
+CONFIG_VIDEOMODE_HELPERS=y
-- 
2.39.2

